tests/test_immutableset.py

```diff
@@ -200,6 +200,11 @@ def test_singleton_index(self):
 
         self.assertEqual(1, s[0])
         self.assertEqual(1, s[-1])
+        with self.assertRaises(IndexError):
```

Comment:
You need a `with` block for each line. This just tests that at least one of these raises an exception.

Reply:
good catch